test/test_launcher.rb

```diff
@@ -79,6 +79,57 @@ def test_pid_file
     File.unlink tmp_path
   end
 
+  def test_state_permission_0640
+    tmp_file = Tempfile.new("puma-state")
+    tmp_path = tmp_file.path
+    tmp_file.close!
+    tmp_permission = 0640
+
+    conf = Puma::Configuration.new do |c|
+      c.state_path tmp_path
+      c.state_permission tmp_permission
+    end
+
+    launcher(conf).write_state
+
+    assert File.stat(tmp_path).mode.to_s(8)[-4..-1], tmp_permission
+
+    File.unlink tmp_path
+  end
+
+  def test_state_permission_nil
+    tmp_file = Tempfile.new("puma-state")
+    tmp_path = tmp_file.path
+    tmp_file.close!
```

Comment:
Same here I think? I think we can just refute File.exist?

Reply:
removed the file closure and implemented @ekohl suggestion to add `ensure`